src/psyclone/tests/psyir/transformations/increase_rank_loop_arrays_test.py

```diff
@@ -0,0 +1,130 @@
+
+
+def test_irla_validate(fortran_reader):
+    ''' Check that the validate method works as expected. '''
+    trans = IncreaseRankLoopArraysTrans()
+
+    psyir = fortran_reader.psyir_from_source("""
+        program test
+            real, dimension(10) :: a
+            integer :: i = 3
+            a(i) = 3
+            do i = 1, 10
+               a(i) = 3
+            end do
+            associate(x => a(i))
+            end associate
+        end program
+    """)
+
+    routine = psyir.children[0]
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[0])
+    assert ("The target of the IncreaseRankLoopArraysTrans transformation "
+            "should be a Loop, but found 'Assignment'." in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1])
+    assert ("IncreaseRankLoopArraysTrans has a mandatory 'arrays' option "
+            "that needs to be provided to inform what arrays needs their "
+            "rank increased." in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1], options={'arrays': ['i']})
+    assert ("IncreaseRankLoopArraysTrans provided 'arrays' must be a local "
+            "array symbols, but 'i: " in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1], options={'arrays': ['a']})
+    assert ("IncreaseRankLoopArraysTrans does not support arrays that are "
+            "referenced inside a Codeblock, but 'a' is inside one."
+            in str(err.value))
+
+    # Referencing non-existan arrays is fine, they are just ignored, this
+    # is useful to apply the transformation to several routines without an
+    # exact match of symbols
+    trans.apply(routine.children[1], options={'arrays': ['non_existant']})
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1].detach(), options={'arrays': ['a']})
+    assert ("The target of the IncreaseRankLoopArraysTrans transformation "
+            "should be a Loop inside a Routine." in str(err.value))
+
+
+def test_irla_apply(fortran_reader, fortran_writer):
+    ''' Check that the array rank is increased by the bounds of the loop. '''
+    trans = IncreaseRankLoopArraysTrans()
+
+    psyir = fortran_reader.psyir_from_source("""
+     program test
+         integer :: N=10, M=10
+         integer :: i, j
+         real, dimension(N) :: ztmp
+         do i = -5, M+3
+             do j = 1, N
+                 ztmp(j) = 1
+             end do
+             do j = 1, N
+                 ztmp(j) = ztmp(j) + 1
+             end do
+         end do
+     end program
+    """)
+
+    trans.apply(psyir.walk(Loop)[0], options={'arrays': ['ztmp']})
+    code = fortran_writer(psyir)
+
+    # The declaration and references have been updated
+    assert "real, dimension(n,-5:m + 3) :: ztmp" in code
```

Comment:
We need a separate test for a case where e.g. `m` is written to before the `i` loop.

Reply:
I think one of the new tests covers this